test/unit/hostgroup_test.rb

```diff
@@ -357,14 +357,41 @@ def setup_user(operation)
     assert_equal "#{hostgroup.id}-a-b",  hostgroup.to_param
   end
 
-  test "clone should clone config groups as well" do
-    group = FactoryGirl.create(:hostgroup, :name => 'a')
-    config_group = ConfigGroup.create!(:name => 'Blah')
-    group.config_groups << config_group
-    group.save
-
-    cloned = group.clone("new_name")
-    assert cloned.config_groups.include?(config_group)
+  context "#clone" do
+    let(:group) {FactoryGirl.create(:hostgroup, :name => 'a')}
+
+    test "clone should clone config groups as well" do
+      config_group = ConfigGroup.create!(:name => 'Blah')
+      group.config_groups << config_group
+      group.save
+
+      cloned = group.clone("new_name")
+      assert cloned.config_groups.include?(config_group)
+    end
+
+    test "clone should clone puppet classes" do
+      group.puppetclasses << FactoryGirl.create(:puppetclass)
+      group.save
+      cloned = group.clone("new_name")
+      assert cloned.puppetclasses.size == 1
+      assert cloned.puppetclasses == group.puppetclasses
+    end
+
+    test "clone should clone parameters" do
+      group.group_parameters.create!(:name => "foo", :value => "bar")
+      cloned = group.clone("new_name")
+      cloned.save!(:validate => false)
```

Comment:
No need to save and hit the database here, `cloned` contains `group_parameters` already at this point.